app/seeders/basic_data/activity_seeder.rb

```diff
@@ -0,0 +1,72 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+module BasicData
+  class ActivitySeeder
+
+    def self.seed!
+      if TimeEntryActivity.any?
```

Comment:
*Needs more OOP*. I.e. I would suggest having a base class for all seeders, making the `seed!` method an instance method and adding a base method for this check if the seeder has to be executed.

Think:

```ruby
module BasicData
  class Seeder
    def seed!
      raise NotImplementedError
    end

    def applicable?
      true
    end
  end
end

# ...

module BasicData
  class ActivitySeeder < Seeder
    def seed!
      TimeEntryActivity.transaction ...
    end
  end

  def applicable?
    TimeEntryActivity.any?.tap do |_|
      puts '   *** Skipping activities as there are already some configured'
    end
  end
end